lib/elixir/src/elixir_def.erl

```diff
@@ -69,13 +69,13 @@ fetch_definition([Tuple | T], File, Module, Set, Bag, All, Private) ->
 
   try ets:lookup_element(Bag, {clauses, Tuple}, 2) of
     Clauses ->
+      MetaCheck = add_check_to_meta(Check, Meta),
```

Comment:
I would prefer to not add a new check flag and rely on the existing ones. In this case, I think we can check for the `:context` flag instead. It is one of the things that we use to set `Check` to true and it should be enough in this case.